packages/catalog-server/src/lib/catalog.ts

```diff
@@ -84,25 +93,27 @@ export class Catalog {
         Temporal.Duration.compare(timeSinceUpdate, packageRefreshInterval) ===
         -1
       ) {
-        return;
+        return {};
       }
       await this.#repository.startPackageUpdate(packageName);
     }
 
     // Fetch package metadata from npm:
+    console.log('Fetching package metadata...');
     let newPackage: Package | undefined;
     try {
       newPackage = await this.#files.getPackageMetadata(packageName);
     } catch (e) {
       await this.#repository.endPackageImportWithError(packageName);
-      return;
+      return {};
     }
+    console.log(' done');
```

Comment:
With concurrency, these "done" logs might be hard to correlate to the action they are associated with.

Reply:
Very true. We need to decide how much we want to log and how to do so. I've never use GCP Logging, but it looks like it has nice analytics features that could come in handy: https://cloud.google.com/logging